src/dev/platform/f4xx/Timerf4xx.cpp

```diff
@@ -0,0 +1,252 @@
+    } else if (peripheral == TIM3) {
+        __HAL_RCC_TIM3_CLK_DISABLE();
+        irqNum = TIM3_IRQn;
+    } else if (peripheral == TIM4) {
+        __HAL_RCC_TIM4_CLK_DISABLE();
+        irqNum = TIM4_IRQn;
+    } else if (peripheral == TIM5) {
+        __HAL_RCC_TIM5_CLK_DISABLE();
+        irqNum = TIM5_IRQn;
+    } else if (peripheral == TIM9) {
+        __HAL_RCC_TIM9_CLK_DISABLE();
+        irqNum = TIM1_BRK_TIM9_IRQn;
+    } else if (peripheral == TIM10) {
+        __HAL_RCC_TIM10_CLK_DISABLE();
+        irqNum = TIM1_UP_TIM10_IRQn;
+    } else if (peripheral == TIM11) {
+        __HAL_RCC_TIM11_CLK_DISABLE();
+        irqNum = TIM1_TRG_COM_TIM11_IRQn;
+    } else if (peripheral == TIM12) {
+        __HAL_RCC_TIM12_CLK_DISABLE();
+        irqNum = TIM8_BRK_TIM12_IRQn;
+    } else if (peripheral == TIM13) {
+        __HAL_RCC_TIM13_CLK_DISABLE();
+        irqNum = TIM8_UP_TIM13_IRQn;
+    } else if (peripheral == TIM14) {
+        __HAL_RCC_TIM14_CLK_DISABLE();
+        irqNum = TIM8_TRG_COM_TIM14_IRQn;
+    } else {
+        return;// Should never reach, but if an invalid peripheral is passed in then simply return
+    }
+
+    HAL_NVIC_DisableIRQ(irqNum);
+}
+
+extern "C" void TIM2_IRQHandler(void) {
+    HAL_TIM_IRQHandler(&halTimers[getTimerInterruptIndex(TIM2)]);
+}
+
+extern "C" void TIM3_IRQHandler(void) {
+    HAL_TIM_IRQHandler(&halTimers[getTimerInterruptIndex(TIM3)]);
+}
+
+extern "C" void TIM4_IRQHandler(void) {
+    HAL_TIM_IRQHandler(&halTimers[getTimerInterruptIndex(TIM4)]);
+}
+
+extern "C" void TIM5_IRQHandler(void) {
+    HAL_TIM_IRQHandler(&halTimers[getTimerInterruptIndex(TIM5)]);
+}
+
+extern "C" void TIM1_BRK_TIM9_IRQHandler(void) {
+    HAL_TIM_IRQHandler(&halTimers[getTimerInterruptIndex(TIM9)]);
+}
+
+extern "C" void TIM1_UP_TIM10_IRQHandler(void) {
+    HAL_TIM_IRQHandler(&halTimers[getTimerInterruptIndex(TIM10)]);
+}
+
+extern "C" void TIM1_TRG_COM_TIM11_IRQHandler(void) {
+    HAL_TIM_IRQHandler(&halTimers[getTimerInterruptIndex(TIM11)]);
+}
+
+extern "C" void TIM8_BRK_TIM12_IRQHandler(void) {
+    HAL_TIM_IRQHandler(&halTimers[getTimerInterruptIndex(TIM12)]);
+}
+
+extern "C" void TIM8_UP_TIM13_IRQHandler(void) {
+    HAL_TIM_IRQHandler(&halTimers[getTimerInterruptIndex(TIM13)]);
+}
+
+extern "C" void TIM8_TRG_COM_TIM14_IRQHandler(void) {
+    HAL_TIM_IRQHandler(&halTimers[getTimerInterruptIndex(TIM14)]);
+}
+
+extern "C" void HAL_TIM_PeriodElapsedCallback(TIM_HandleTypeDef* htim) {
+    uint8_t interruptIdx = getTimerInterruptIndex(htim->Instance);
+    if (timerInterruptHandlers[interruptIdx] != nullptr) {
+        timerInterruptHandlers[interruptIdx](htim);
+    }
+}
+
+/**
+ * Returns the corresponding index for the given timer device as stored in the
+ * timerInterruptHandlers array
+ * @param htim the TIM_TypeDef pointer to the timer device
+ * @return an uint8_t between 0-9 corresponding to an element of timerInterruptHandlers
+ */
+uint8_t getTimerInterruptIndex(TIM_TypeDef* peripheral) {
```

Comment:
Turn this into a switch statement in my opinion, too many else if's

Reply:
The peripheral that is being if'ed on is a memory address for the timer, which as far as i know cannot be used in a switch (i think)

Reply:
Are you saying with `enum` instead of `enum class` I don't need to do the static cast to uint8, or are you saying something else is possible with `enum` instead of `enum class`?

Reply:
> Are you saying with `enum` instead of `enum class` I don't need to do the static cast to uint8

Yes, this is exactly what I'm saying. You can pass an `enum` into a function asking for `uint8_t`